### ISSUE TITLE
Allow scm-source to be executed outside of the source tree

### ISSUE THREAD

**Issue by lmineiro** [zalando-stups/python-scm-source#1]
For some build environments it's not easy or possible to execute scm-source while being in the source tree folder.

For ex., it would be possible to run:

`make -f /path/to/project/Makefile`

And a possible target could be `deploy` that generated the scm-source.json

Currently the tool always expected to be executed inside the source tree and there is no option to execute against different paths.

I suggest a new (optional) argument that is the path where the source tree is. For ex.

`scm-source -f $CODE_ROOT/deploy/scm-source.json $CODE_ROOT`

**Comment by hjacobs** [comment]
yes, I agree --- I just wanted to have a first hack released, so I did not implement it from the start :smile:

**Comment by hjacobs** [comment]
I think it should be an option, as it is "optional" --- e.g. "-d" for directory

**Comment by hjacobs** [comment]
or -s/--source ?

**Comment by lmineiro** [comment]
My idea was that this tool does exactly one thing - it generates the file from a source tree. Its single purpose, its "object", is actually a source tree. Just like "ls", for ex., it lists the current folder if you don't specify that argument, but you can, in which case it will list the folder specified as the argument.

For the principle of least surprise, I'd expect it to accept that argument, and not a specific option.

The existing "-f" I'd expect to be -o, actually, since if controls the output of what it does.

More important than the cosmetics it would be the feature though. Our Makefile is a bit ugly now with some pushd's and popd's

**Comment by lmineiro** [comment]
For the shell script variant I found this doable using:

```
REV=$(git --git-dir=${CWD}/.git rev-parse HEAD)
URL=$(git --git-dir=${CWD}/.git config --get remote.origin.url)
STATUS=$(git --git-dir=${CWD}/.git --work-tree=${CWD} status --porcelain)
```

Where CWD could be the new argument, defaulting to `.`
